### ISSUE TITLE
2.15.8 breaks nanoc

### ISSUE THREAD

**Issue by ashmckenzie** [ddfreyne/cri#104]
When using 2.15.8+ with nanoc and a port is specifed, nanoc errors out:

```ruby
$ bundle exec nanoc live -p 3005
#<Thread:0x00007fda5490b380@/Users/ash/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/guard-nanoc-2.1.6/lib/guard/../nanoc/cli/commands/live.rb:22 run> terminated with exception (report_on_exception is true):
/Users/ash/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/nanoc-4.11.5/lib/nanoc/cli/commands/view.rb:34:in `fetch': key not found: :host (KeyError)
	from /Users/ash/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/nanoc-4.11.5/lib/nanoc/cli/commands/view.rb:34:in `run'
	from /Users/ash/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/guard-nanoc-2.1.6/lib/guard/../nanoc/cli/commands/live.rb:26:in `block in run'

Captain! We’ve been hit!

KeyError: key not found: :host

  0. /Users/ash/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/nanoc-4.11.5/lib/nanoc/cli/commands/view.rb:34:in `fetch'
  1. /Users/ash/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/nanoc-4.11.5/lib/nanoc/cli/commands/view.rb:34:in `run'
  2. /Users/ash/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/guard-nanoc-2.1.6/lib/guard/../nanoc/cli/commands/live.rb:26:in `block in run'

A detailed crash log has been written to ./crash.log.
```

Looking into the problem, I created the following which takes the Cri defintions straight from https://github.com/nanoc/nanoc/blob/4.11.5/nanoc/lib/nanoc/cli/commands/view.rb#L3-L15

**nanoc_example.rb**

```ruby
#!/usr/bin/env ruby

require 'cri'
require 'nanoc'
require 'nanoc/cli'

command = Cri::Command.define do
  summary 'start the web server that serves static files'
  description <<~EOS
    Start the static web server. Unless specified, the web server will run on port
    3000 and listen on all IP addresses. Running this static web server requires
    `adsf` (not `asdf`!).
  EOS

  required :H, :handler, 'specify the handler to use (webrick/mongrel/...)'
  required :o, :host,    'specify the host to listen on (default: 127.0.0.1)', default: '127.0.0.1'
  required :p, :port,    'specify the port to listen on (default: 3000)', transform: Nanoc::CLI::Transform::Port, default: 3000
  flag :L, :'live-reload', 'reload on changes'
  no_params

  run do |opts, args|
    p opts.fetch(:port)
  end
end

command.run(ARGV)
```

**Gemfile**

```ruby
source "https://rubygems.org"

gem 'cri'
gem 'nanoc'
```

**2.15.7**

```sh
$ bundle exec ./nanoc_example.rb
3000
```

**2.15.9**

```sh
$ bundle exec ./nanoc_example.rb
bundler: failed to load command: ./nanoc_example.rb (./nanoc_example.rb)
KeyError: key not found: :port
  /private/tmp/nanoc_example.rb:22:in `fetch'
  /private/tmp/nanoc_example.rb:22:in `block (2 levels) in <top (required)>'
  /Users/ash/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cri-2.15.9/lib/cri/command.rb:360:in `run_this'
  /Users/ash/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cri-2.15.9/lib/cri/command.rb:296:in `run'
  /private/tmp/nanoc_example.rb:26:in `<top (required)>'
```

Edit: Updated to reflect issue is present since 2.15.8+ and final example was for 2.15.9.

**Comment by ddfreyne** [comment]
Could you upgrade to 2.15.9? 2.15.8 did indeed have this bug.

If this issue is present in 2.15.9 as well, do reopen this issue.

**Comment by ashmckenzie** [comment]
Whilst I referred to 2.15.8 at the top, this bug exists for 2.15.8+, including 2.15.9 as I was using 2.15.9 for the test:

```sh
$ grep cri Gemfile.lock
    cri (2.15.9)
      cri (~> 2.15)
  cri
```

I would re-open if I could 😄

**Comment by Fjan** [comment]
Just a quick note to add that `nanoc view` also breaks on cri 2.15.9, with the same error

**Comment by ddfreyne** [comment]
This is the same as https://github.com/nanoc/nanoc/issues/1441 -- it is fixed on `master`, but unfortunately I do not have a functioning computer with me to do the release. Next week!

**Comment by ddfreyne** [comment]
(on master of nanoc, not master of cri)

**Comment by ddfreyne** [comment]
A fix for this is released in Nanoc 4.11.6.
